docs/admin/setup.mdx

```diff
@@ -0,0 +1,137 @@
+│   │   │   └── genesis.json
+│   │   └── data
+│   ├── config.toml
+│   └── private_key
+├── node1
+│   ├── abci
+│   │   ├── config
+│   │   │   └── genesis.json
+...
+```
+
+The config files for each of the nodes will specify all of the other nodes a
+persistent peers so that they will connect to each other on startup. This is
+generally only practical for small test networks with fewer than 12 nodes.
+
+The following options may be used to control what is generated:
+
+- `--validators` or `-v` is number of validators [default: 4]
+- `--non-validators` or `-n` is the number of non-validators [default: 4]
+- `--config` is a template config file to use, default is none
+- `--output-dir` or `-o` is the parent directory for all of generated node folders [default: ".testnet"]
+- `--node-dir-prefix` is the prefix for the node directories (node results in node0, node1, ...) [default: "node"]
+- `--hostname-prefix` is the prefix for node host names e.g. node results in node0, node1, etc.
+- `--hostname-suffix` is the suffix for node host names e.g. .example.com results in node0.example.com, node1.example.com, etc.
+- `--starting-ip` is the starting IP address of the first network node, with subsequent nodes on the next address in the subnet
+- `--hostname` overrides all hostnames of the nodes
+- `--p2p-port` or `-p` is the TCP port port on which the nodes listens for P2P connections
+
+## Updating Genesis Config with Initial SQLite Data
+
+If preparing a network with initial databases, it is required to update the "app
+hash" in `kwild`'s genesis configuration to reflect the initial state that
+includes one or more SQLite files. The syntax is:
+
+```
+kwil-admin setup genesis-hash [--genesis GENESIS] [DBDIR]
+```
+
+Both inputs are optional, using default paths. `GENESIS` is the path to the
+genesis file to patch, which is `~/.kwild/abci/config/genesis.json` by default.
+The `DBDIR` argument is the directory containing all of kwild's .sqlite files to
+be hashed, which is `~/.kwild/data/kwil.db/` by default.
+
+## Resetting All Application Data
+
+To delete all of a Kwil node's data files, use the `reset` command.
+
+**WARNING: This command should not be used on production systems. This should
+only be used to reset disposable test nodes.**
+
+```
+kwil-admin setup reset [--root_dir DIR] [--force] [--sqlpath DIR] [--snappath DIR]
+```
+
+This command only requires one: (a) `--root_dir` to specify a node root
+directory containing the files to reset, or (b) `--force` to automatically reset
+the files in the default node root directory at `~/.kwild`.
+
+The paths to the SQLite files (`--sqlpath`) and snapshots (`--snappath`) are
+optional, but provided here since they are configurable on the node.
+
+## Resetting Only Chain state
+
+To reset the block chain state data while leaving all of the other application
+data such as the SQLite databases intact, use the `reset-state` command. This
+will delete the data files under `<root_dir>/abci/data`
```

Comment:
@charithabandi I actually wasn't terribly clear on the use case for `reset-state`.  Would you add or change the description here?

Reply:
This is more for the development purposes and not to be used in the production. 

It basically resets the chain state, that means it deletes any data generated by the wal like mempool wal, block data, state data etc.  So everything under `<root_dir>/abci/data` and `<root_dir>/abci/info`

Reply:
What I'm getting at is the specific development workflows where `reset-state` would be used.  Do you use it regularly?  If you delete state but not the rest of the application data, what then?  Is that where you'd use the `genesis-hash` subcommand?

I'm a bit hesitant to put commands in kwil-admin that are for development only, particularly if I don't see the workflow that would demand it.

Reply:
Yeah, reset-state can be removed, we use reset-all for more of our testing